src/cmd_line/commands/substitute.ts

```diff
@@ -79,15 +79,16 @@ export class SubstituteCommand extends node.CommandBase {
       jsRegexFlags += 'i';
     }
 
-    // If no pattern is entered, use previous search state
+    // If no pattern is entered, use previous search state (including search with * and #)
     if (args.pattern === '') {
-      const prevSearchState = vimState.globalState.searchStatePrevious;
-      if (prevSearchState) {
-        const prevSearchString = prevSearchState[prevSearchState.length - 1].searchString;
-        args.pattern = prevSearchString;
+      const prevSearchState = vimState.globalState.searchState;
+      if (prevSearchState === undefined || prevSearchState.searchString === '') {
+        // Should we use VimError (in src/error.ts) ?
+        throw new Error('Empty search string!');
```

Comment:
Can you implement and throw the `VimError`?

Reply:
I tried, but the handling of `VimError` is either broken or, more likely, I don't understand how it works. 

Whenever a `VimError` is thrown, `StatusBarImpl.setText` is invoked with the text given to `VimError` (commandLine.ts), but is then immediately invoked again with "`-- NORMAL -- :`" and again with "`-- NORMAL -- `" as text. Therefore, the user does not get a chance to see the error thrown by VimError. 

![animated](https://user-images.githubusercontent.com/812924/38499023-9c662948-3c06-11e8-9fd8-a9970aef7763.gif)